Makefile

```diff
@@ -8,268 +27,508 @@ usage:
+		brew install libusb-compat pkg-config;\
+	else\
+		echo "";\
+		echo "ERR: Do not know how to install libusb on this platform.";\
+		exit 1;\
+	fi
+endef
+
+.PHONY: ci-setup-tools
+ci-setup-tools:
+	$(call ci_setup_helper,\
+		pkg-config --cflags --libs libusb &> /dev/null && echo yes,\
+		Install 'libusb' for development using your package manager,\
+		ci_setup_tools,\
+		CI_JOB_TOOLS)
+
+define ci_job_tools
+	$(call banner,CI-Job: Tools)
+	@for tool in `./tools/list_tools.sh`;\
+		do echo "$$(tput bold)Build & Test $$tool";\
+		cd tools/$$tool;\
+		CI=true RUSTFLAGS="-D warnings" cargo build --all-targets || exit 1;\
+		cd - > /dev/null;\
+		done
+endef
+
+.PHONY: ci-job-tools
+ci-job-tools: ci-setup-tools
+	$(if $(CI_JOB_TOOLS),$(call ci_job_tools))
+
+
+
+### ci-runner-github-qemu jobs:
+
+define ci_setup_qemu_riscv
+	@# Use the latest QEMU as it has OpenTitan support
+	@printf "Building QEMU, this could take a few minutes\n\n"
+	# Download Tock qemu fork if needed
+	if ! bash -c 'cd tools/qemu && [[ $$(git rev-parse --short HEAD) == "7ff5b84" ]]'; then \
 		rm -rf tools/qemu; \
 		cd tools; git clone https://github.com/alistair23/qemu.git --depth 1 -b riscv-tock.next; \
 		cd qemu; ./configure --target-list=riscv32-softmmu; \
 	fi
+	# Build qemu
 	@$(MAKE) -C "tools/qemu" || (echo "You might need to install some missing packages" || exit 127)
+endef
+
+define ci_setup_qemu_opentitan
+	# Download OpenTitan image
 	@printf "Downloading OpenTitan boot rom from: 2aedf641120665b91c3a5d5aa214175d09f71ee6\n"
-	$(eval CURRENT_DIR := $(shell pwd))
-	@pushd `mktemp -d -t`; \
-		curl `curl "https://dev.azure.com/lowrisc/opentitan/_apis/build/builds/13066/artifacts?artifactName=opentitan-dist&api-version=5.1" | cut -d \" -f 38` --output opentitan-dist.zip; \
+	@pwd=$$(pwd) && \
+		temp=$$(mktemp -d)\
+		cd $$temp && \
+		curl $$(curl "https://dev.azure.com/lowrisc/opentitan/_apis/build/builds/13066/artifacts?artifactName=opentitan-dist&api-version=5.1" | cut -d \" -f 38) --output opentitan-dist.zip; \
 		unzip opentitan-dist.zip; \
-		tar  -xf opentitan-dist/opentitan-snapshot-20191101-*.tar.xz; \
-		mv opentitan-snapshot-20191101-*/sw/device/boot_rom/boot_rom_fpga_nexysvideo.elf $(CURRENT_DIR)/opentitan-boot-rom.elf
+		tar -xf opentitan-dist/opentitan-snapshot-20191101-*.tar.xz; \
+		mv opentitan-snapshot-20191101-*/sw/device/boot_rom/boot_rom_fpga_nexysvideo.elf $$pwd/tools/qemu-runner/opentitan-boot-rom.elf
+endef
 
+.PHONY: ci-setup-qemu
+ci-setup-qemu:
+	$(call ci_setup_helper,\
+		cd tools/qemu && [[ $$(git rev-parse --short HEAD) == "1ef6d40" ]] && [ -x riscv32-softmmu ] && echo yes,\
+		Clone QEMU fork (with riscv fixes) and run its build scripts,\
+		ci_setup_qemu_riscv,\
+		CI_JOB_QEMU_RISCV)
+	$(call ci_setup_helper,\
+		[[ $$(cksum tools/qemu-runner/opentitan-boot-rom.elf | cut -d' ' -f1) == "2835238144" ]] && echo yes,\
+		Download opentitan archive and unpack a ROM image,\
+		ci_setup_qemu_opentitan,\
+		CI_JOB_QEMU_OPENTITAN)
```

Comment:
Same here, nothing is being installed.

Reply:
Same answer I think, I read "install" here as "install bits onto disk":

Tock needs to install something to run this step.

> You have run a (likely CI) rule that requires Tock to install something on your
> machine. Tock can do this automatically for you, or you can look at the recipe
> for 'ci_setup_qemu_opentitan' and install yourself.
> 
> Continuing will: Download opentitan archive and unpack a ROM image.
> 
> You can use 'make ci-nosetup' to run all CI with no new install requirements.
> 
> Should Tock run install steps for you? [y/N] ^C

I think the "Continuing will" line conveys the intent well. Do you have suggestions for what you think would be better?